File: Homework4/Question3/src/Jellyfish.java
public class Jellyfish implements Swimmer {
    private String name;

    public Jellyfish(String name) {
        this.name = name;
    }

    @Override
    public String getName() {
        return name;
    }

    @Override
    public Swimmer getFriend() {
        System.out.println("The friend of Jellyfish is itself.");
        return this;
    }

    public static void testJellyFish() {
        Jellyfish j = new Jellyfish("Blob");
        System.out.println(j.getName() == "Blob");
        //The friend of the jellyfish is the jellyfish itself
        System.out.println(j.getFriend() == j);
    }
}

File: Homework4/Jellyfish.kt
class Jellyfish : Swimmer {
    private val name: String

    constructor(name: String) {
        this.name = name
    }

    override fun getName(): String {
        return name
    }

    override fun getFriend(): Swimmer {
        println("The friend of Jellyfish is itself.")
        return this
    }

    companion object {
        fun testJellyfish() {
            val j = Jellyfish("Blob")
            println(j.getName() == "Blob")
            println(j.getFriend() == j)
        }
    }
}